lib/asciidoctor/pdf/converter.rb

```diff
@@ -933,9 +933,17 @@ def convert_open node
       # TODO process abstract child even when partintro has multiple blocks
       convert_abstract node.blocks[0]
     else
-      add_dest_for_block node if node.id
-      layout_caption node.title if node.title?
-      convert_content_for_block node
+      if node.has_role? 'keep-together'
+        keep_together do
```

Comment:
We can make this DRY by using `keep_together_if`, like:

```rb
keep_together_if node.option? 'unbreakable' do
```

To avoid footnotes from being duplicated, it's also necessary to add this line at the start of the block:

```rb
push_scratch doc if scratch?
```

and this line to the end:

```rb
pop_scratch doc if scratch?
```

In all, that looks like:

```rb
doc = node.document
keep_together_if node.option? 'unbreakable' do
  push_scratch doc if scratch?
  add_dest_for_block node if node.id
  layout_caption node.title if node.title?
  convert_content_for_block node
  pop_scratch doc if scratch?
end
```

Reply:
Thanks for your review.
I wasn't aware of this issue with footnotes. I will add footnotes in my test to make sure that they are not duplicated.

Reply:
It's a bit weird, and I wish it was handled by keep_together. But that would require some refactoring. For now, we just add it where keep_together is used.